jwql/website/apps/jwql/archive_database_update.py

```diff
@@ -124,25 +126,22 @@ def get_updates(update_database):
             elif len(metadata_proprietary['category']):
                 proposal_category = metadata_proprietary['category'][0]
 
-            # Get set of unique rootnames
-            all_rootnames = set(['_'.join(f.split('/')[-1].split('_')[:-1]) for f in filenames])
+            # Get set of unique rootnames and file paths
+            seen = set()
+            all_rootnames = []
+            all_filenames = []
+            indices = []
 
-            # Filter source-based level 2 files out of the rootnames and filenames
-            all_rootnames = filter_rootnames(all_rootnames)
-            filenames = filter_filenames(filenames, all_rootnames)
+            for i, item in enumerate(filenames):
+                rname = '_'.join(item.split('/')[-1].split('_')[:-1])
+                if rname not in seen:
+                    seen.add(rname)
```

Comment:
This is cool, overriding the `set()` method to make it make sense for our context.

Reply:
"seen" is just an empty set to start with. Then entries are added to complete a set of filenames